community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/ConstraintIndexCreator.java

```diff
@@ -131,48 +120,31 @@ public long createUniquenessConstraintIndex(
         Client locks = state.locks().pessimistic();
         try
         {
-            long indexId = schemaOps.indexGetCommittedId( state, index );
-            IndexProxy proxy = indexingService.getIndexProxy( indexId );
-
             // Release the LABEL WRITE lock during index population.
             // At this point the integrity of the constraint to be created was checked
             // while holding the lock and the index rule backing the soon-to-be-created constraint
             // has been created. Now it's just the population left, which can take a long time
-            monitor.event( Thread.currentThread().getId(), Monitor.RELEASING_LOCK );
             releaseLabelLock( locks, descriptor.getLabelId() );
-            monitor.event( Thread.currentThread().getId(), Monitor.RELEASED_LOCK );
 
-            monitor.event( Thread.currentThread().getId(), Monitor.AWAITED_POPULATION );
-            awaitConstrainIndexPopulation( constraint, proxy );
+            awaitConstrainIndexPopulation( constraint, indexConstraintState.indexProxy );
 
             // Index population was successful, but at this point we don't know if the uniqueness constraint holds.
             // Acquire LABEL WRITE lock and verify the constraints here in this user transaction
             // and if everything checks out then it will be held until after the constraint has been
             // created and activated.
-            monitor.event( Thread.currentThread().getId(), Monitor.AWAITING_LOCK );
             acquireLabelLock( state, locks, descriptor.getLabelId() );
-            monitor.event( Thread.currentThread().getId(), Monitor.GOT_LOCK );
             reacquiredLabelLock = true;
 
-            indexingService.getIndexProxy( indexId ).verifyDeferredConstraints( propertyAccessor );
+            indexConstraintState.indexProxy.verifyDeferredConstraints( propertyAccessor );
+            indexConstraintState.indexProxy.activate();
```

Comment:
I would have thought that activation could only be done by transaction applier, not willy-nilly out in kernel code...